envi/archs/ppc/__init__.py

```diff
@@ -75,25 +76,64 @@ def __init__(self, mode=64, archname='ppc-embedded', endian=envi.ENDIAN_MSB):
         self.psize = mode//8
         self.maps = tuple()
         if self.psize == 8:
-            self._arch_dis = Ppc64EmbeddedDisasm()
+            self._arch_dis = eapd.Ppc64EmbeddedDisasm()
         else:
-            self._arch_dis = Ppc32EmbeddedDisasm()
+            self._arch_dis = eapd.Ppc32EmbeddedDisasm()
         self._arch_vle_dis = vle.VleDisasm()
 
         # used to store VLE information
-        # Default the alignment (page size) to be the entire valid memory range
-        self._page_size = 2 ** mode
+        # Default the alignment (page size) to the default envi PAGE_SIZE
+        self._page_size = e_const.PAGE_SIZE
         self._page_mask = 0
         self._vle_pages = {}
 
+    def getPpcVleInfoSnap(self):
+        snap = self._page_size, self._page_mask, self._vle_pages
+        return snap
+
+    def setPpcVleInfoSnap(self, snap):
+        self._page_size, self._page_mask, self._vle_pages = snap
+
+    def getEmuSnap(self):
+        """
+        Return the data needed to "snapshot" this emulator.  For PPC we need to
+        include the PPC VLE memory map info here.
+        """
+        regs = self.getRegisterSnap()
+        mem = self.getMemorySnap()
+        vleinfo = self.getPpcVleInfoSnap()
+        return regs, mem, {'vlemaps': vleinfo}
+
+    def setEmuSnap(self, snap):
+        # Check if the snap has extra information, otherwise assume this is the 
+        # standard two snap elements
+        if len(snap) == 3:
+            regs, mem, sdict = snap
+            vleinfo = sdict.get('vlemaps')
+        else:
+            regs, mem = snap
+            vleinfo = None
+
+        self.setRegisterSnap(regs)
+        self.setMemorySnap(mem)
+
+        if vleinfo is not None:
+            self.setPpcVleInfoSnap(vleinfo)
```

Comment:
we need to plumb this version of `get/setEmuSnap()` through all the architectures

Reply:
does it need to be made in everywhere? In theory if the architecture doesn't use the 3rd item it should never be generated, right?

Reply:
it's an api that is standard across all arches.  while we could probably get away with a black-box of snapshotting the various emulators, that's not preferred.  currently there's an assumption of consistency.